python/lsst/dax/apdb/tests/_apdb.py

```diff
@@ -392,6 +396,34 @@ def test_storeForcedSources(self) -> None:
         res = apdb.containsVisitDetector(visit=2, detector=42)
         self.assertFalse(res)
 
+    def test_timestamps(self) -> None:
+        """Check that timestamp return type is as expected."""
+        config = self.make_instance()
+        apdb = Apdb.from_config(config)
+
+        region = _make_region()
+        visit_time = self.visit_time
+
+        # have to store Objects first
+        time_before = datetime.datetime.now()
+        objects = makeObjectCatalog(region, 100, visit_time)
+        oids = list(objects["diaObjectId"])
+        catalog = makeForcedSourceCatalog(objects, visit_time)
+        time_after = datetime.datetime.now()
+
+        apdb.store(visit_time, objects, forced_sources=catalog)
+
+        # read it back and check sizes
+        res = apdb.getDiaForcedSources(region, oids, visit_time)
+        assert res is not None
+        self.assert_catalog(res, len(catalog), ApdbTables.DiaForcedSource)
+
+        self.assertIn("time_processed", res.dtypes)
+        dtype = res.dtypes["time_processed"]
+        self.assertEqual(dtype.name, self.timestamp_type_name)
+        # Verify that returned time is sensible.
+        self.assertTrue(all(time_before <= dt <= time_after for dt in res["time_processed"]))
```

Comment:
Doesn't this basically amount to testing `makeObjectCatalog` and `makeForcedSourceCatalog` (i.e., test functions)? In production, setting these timestamps is the responsibility of `ap_association`.

Reply:
This tests that round-tripping of timestamps works reasonably, I was worried that saving/restoring may add timezone. This is not a perfect test, I'll think about it on the next ticket.